subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -118,31 +141,105 @@ static bool zero_id_check(uint8_t start_idx)
 	return true;
 }
 
-static bool id_sequence_check(uint8_t start_idx)
+static bool id_sequence_check(uint8_t stop_idx)
 {
-	__ASSERT_NO_MSG(start_idx > 0);
-	for (size_t i = start_idx; i < ACCOUNT_KEY_CNT; i++) {
-		if (account_key_loaded_ids[i] !=
-		    next_account_key_id(account_key_loaded_ids[i - 1])) {
+	for (size_t i = 0; i < stop_idx; i++) {
+		if (account_key_loaded_ids[i] != ACCOUNT_KEY_MIN_ID + i) {
 			return false;
 		}
 	}
 
 	return true;
 }
 
-static bool rollover_check(uint8_t cur_id, uint8_t prev_id)
+static uint8_t toggle_ak_id(uint8_t id)
 {
-	return ((cur_id == (prev_id - (ACCOUNT_KEY_CNT - 1))) ||
-		(cur_id == (prev_id + (ACCOUNT_KEY_CNT + 1))));
+	__ASSERT_NO_MSG(id >= ACCOUNT_KEY_MIN_ID);
+
+	return (id < ACCOUNT_KEY_MIN_ID + ACCOUNT_KEY_CNT) ? (id + ACCOUNT_KEY_CNT) :
+							     (id - ACCOUNT_KEY_CNT);
+}
+
+uint8_t next_account_key_id(void)
+{
+	if (account_key_count < ACCOUNT_KEY_CNT) {
+		return ACCOUNT_KEY_MIN_ID + account_key_count;
+	}
+
+	return toggle_ak_id(account_key_order[ACCOUNT_KEY_CNT - 1]);
+}
+
+static void ak_order_update(uint8_t id)
+{
+	bool id_found = false;
+	size_t found_idx;
+
+	for (size_t i = 0; i < account_key_count; i++) {
+		if (account_key_order[i] == id) {
+			id_found = true;
+			found_idx = i;
+			break;
+		}
+	}
+
+	if (id_found) {
+		for (size_t i = found_idx; i > 0; i--) {
+			account_key_order[i] = account_key_order[i - 1];
+		}
+	} else {
+		for (size_t i = account_key_count - 1; i > 0; i--) {
+			account_key_order[i] = account_key_order[i - 1];
+		}
+	}
+	account_key_order[0] = id;
+}
+
+static int commit_ak_order(void)
+{
+	int err;
+	size_t ak_order_update_count = 0;
+
+	for (size_t i = account_key_count; i < ACCOUNT_KEY_CNT; i++) {
+		if (account_key_order[i] != 0) {
+			return -EINVAL;
+		}
+	}
+
+	for (size_t i = 0; i < account_key_count; i++) {
+		uint8_t id = account_key_loaded_ids[i];
+		bool id_found = false;
+
+		for (size_t j = 0; j < account_key_count; j++) {
+			if (account_key_order[j] == id) {
+				id_found = true;
+				break;
+			}
+		}
+		if (!id_found) {
+			__ASSERT_NO_MSG(ak_order_update_count < account_key_count);
```

Comment:
In theory, the assert is impossible to trigger and it can be opt out in release builds, but maybe it will be safer to add return here as it don't cost much.

Reply:
Break does not cost much and it allows application to get informed about an error via the settings_load return value. I think it's worth it :)